### ISSUE TITLE
Prometheus is not scrapping custom metric

### ISSUE THREAD

**Issue by montenegrodr** [coreos/prometheus-operator#2622]
**What did you do?**

Hi, thanks for maintaining this tool. I am setting up prometheus-operator in my cluster. But for one specific deployment/service the metrics are not reaching Prometheus. The metrics are from sidekiq, and they look like (**and I have the feeling the error is here, I guess prometheus-operator expect another format?**):

```
# HELP sidekiq_processed_jobs_total The total number of processed jobs.
# TYPE sidekiq_processed_jobs_total counter
sidekiq_processed_jobs_total 121747110

# HELP sidekiq_failed_jobs_total The total number of failed jobs.
# TYPE sidekiq_failed_jobs_total counter
sidekiq_failed_jobs_total 30232127

# HELP sidekiq_workers The number of workers across all the processes.
# TYPE sidekiq_workers gauge
sidekiq_workers 1160
```
**Did you expect to see some different?**

I wanted to see my `sidekiq` metrics in Prometheus:9090. 

- **I know Prometheus Operator is working because it is scrapping other custom metrics from another service/service monitor - the example-app is working in this same cluster**
- **Since those metrics have `http_request` in the name, my guess is I need to follow some strict format in order to Prometheus scrape my custom metrics**

**Environment**

* Prometheus Operator version:

    `Insert image tag or Git SHA here`
     ```
     v0.30.0
     ```
    

* Kubernetes version information:

    `kubectl version`
    
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T16:23:09Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.7-gke.10", GitCommit:"8d9b8641e72cf7c96efa61421e87f96387242ba1", GitTreeState:"clean", BuildDate:"2019-04-12T22:59:24Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:

    Terraform 12

* Manifests:

The `service` from k8s:

```
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2019-06-01T10:58:04Z"
  labels:
    tier: prometheus-exporter
  name: MYAPP-core-qa-MYAPP-prometheus-exporter
  namespace: MYAPP
  resourceVersion: "5131785"
  selfLink: /api/v1/namespaces/MYAPP/services/MYAPP-core-qa-MYAPP-prometheus-exporter
  uid: 214fc3b6-845c-11e9-89ec-42010a8000aa
spec:
  clusterIP: 10.86.14.215
  ports:
  - name: prom-exp
    port: 9292
    protocol: TCP
    targetPort: 9292
  selector:
    app: MYAPP
    release: MYAPP-core-qa
    tier: prometheus-exporter
    version: "2.61"
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

The `service monitor` from k8s:

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  creationTimestamp: "2019-06-01T10:58:04Z"
  generation: 1
  labels:
    release: MYAPP-core-qa
  name: MYAPP-core-qa-MYAPP-sidekiq-admin-service-monitor
  namespace: MYAPP
  resourceVersion: "5131885"
  selfLink: /apis/monitoring.coreos.com/v1/namespaces/MYAPP/servicemonitors/MYAPP-core-qa-MYAPP-sidekiq-admin-service-monitor
  uid: 219cc699-845c-11e9-89ec-42010a8000aa
spec:
  endpoints:
  - port: prom-exp
  selector:
    matchLabels:
      tier: prometheus-exporter
```
The scrapping config from Prometheus:9090/config:

```
- job_name: MYAPP/MYAPP-core-qa-MYAPP-sidekiq-admin-service-monitor/0
  honor_timestamps: true
  scrape_interval: 30s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  kubernetes_sd_configs:
  - role: endpoints
    namespaces:
      names:
      - MYAPP
  relabel_configs:
  - source_labels: [__meta_kubernetes_service_label_tier]
    separator: ;
    regex: prometheus-exporter
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_endpoint_port_name]
    separator: ;
    regex: prom-exp
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_endpoint_address_target_kind, __meta_kubernetes_endpoint_address_target_name]
    separator: ;
    regex: Node;(.*)
    target_label: node
    replacement: ${1}
    action: replace
  - source_labels: [__meta_kubernetes_endpoint_address_target_kind, __meta_kubernetes_endpoint_address_target_name]
    separator: ;
    regex: Pod;(.*)
    target_label: pod
    replacement: ${1}
    action: replace
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_name]
    separator: ;
    regex: (.*)
    target_label: service
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: pod
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_name]
    separator: ;
    regex: (.*)
    target_label: job
    replacement: ${1}
    action: replace
  - separator: ;
    regex: (.*)
    target_label: endpoint
    replacement: prom-exp
    action: replace
```
* Prometheus Operator Logs:

```
ts=2019-06-01T10:58:06.178462234Z caller=main.go:169 msg="Starting Prometheus Operator version '0.30.0'."
level=info ts=2019-06-01T10:58:06.208344713Z caller=operator.go:391 component=prometheusoperator msg="connection established" cluster-version=v1.12.7-gke.10
level=info ts=2019-06-01T10:58:06.208346309Z caller=operator.go:213 component=alertmanageroperator msg="connection established" cluster-version=v1.12.7-gke.10
level=info ts=2019-06-01T10:58:06.540219202Z caller=operator.go:635 component=alertmanageroperator msg="CRD updated" crd=Alertmanager
level=info ts=2019-06-01T10:58:06.576202923Z caller=operator.go:1478 component=prometheusoperator msg="CRD updated" crd=Prometheus
level=info ts=2019-06-01T10:58:06.597018509Z caller=operator.go:1478 component=prometheusoperator msg="CRD updated" crd=ServiceMonitor
level=info ts=2019-06-01T10:58:06.6179536Z caller=operator.go:1478 component=prometheusoperator msg="CRD updated" crd=PrometheusRule
level=info ts=2019-06-01T10:58:09.560748206Z caller=operator.go:229 component=alertmanageroperator msg="CRD API endpoints ready"
level=info ts=2019-06-01T10:58:09.764573303Z caller=operator.go:184 component=alertmanageroperator msg="successfully synced all caches"
level=info ts=2019-06-01T10:58:16.664652175Z caller=operator.go:407 component=prometheusoperator msg="CRD API endpoints ready"
level=info ts=2019-06-01T10:58:23.53446398Z caller=operator.go:342 component=prometheusoperator msg="successfully synced all caches"
level=info ts=2019-06-01T10:58:23.817964008Z caller=operator.go:985 component=prometheusoperator msg="sync prometheus" key=MYAPP/MYAPP-core-qa-prometheus-prometheus
level=info ts=2019-06-01T10:58:38.958182094Z caller=operator.go:985 component=prometheusoperator msg="sync prometheus" key=MYAPP/MYAPP-core-qa-prometheus-prometheus
level=info ts=2019-06-01T10:58:46.71738197Z caller=operator.go:985 component=prometheusoperator msg="sync prometheus" key=MYAPP/MYAPP-core-qa-prometheus-prometheus
level=info ts=2019-06-01T10:59:35.638985932Z caller=operator.go:985 component=prometheusoperator msg="sync prometheus" key=MYAPP/MYAPP-core-qa-prometheus-prometheus
W0601 11:03:41.590009       1 reflector.go:289] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:417: watch of *v1.ConfigMap ended with: too old resource version: 5132067 (5133097)
level=info ts=2019-06-01T11:06:01.695114684Z caller=operator.go:985 component=prometheusoperator msg="sync prometheus" key=MYAPP/MYAPP-core-qa-prometheus-prometheus
level=info ts=2019-06-01T11:06:01.755507916Z caller=operator.go:985 component=prometheusoperator msg="sync prometheus" key=MYAPP/MYAPP-core-qa-prometheus-prometheus
W0601 11:11:16.609252       1 reflector.go:289] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:417: watch of *v1.ConfigMap ended with: too old resource version: 5133507 (5135114)
W0601 11:18:05.633946       1 reflector.go:289] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:417: watch of *v1.ConfigMap ended with: too old resource version: 5135518 (5136928)
W0601 11:23:14.651876       1 reflector.go:289] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:417: watch of *v1.ConfigMap ended with: too old resource version: 5137333 (5138298)
W0601 11:29:45.662979       1 reflector.go:289] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:417: watch of *v1.ConfigMap ended with: too old resource version: 5138701 (5140030)
W0601 11:36:34.680167       1 reflector.go:289] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:417: watch of *v1.ConfigMap ended with: too old resource version: 5140432 (5141837)
W0601 11:43:08.698050       1 reflector.go:289] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:417: watch of *v1.ConfigMap ended with: too old resource version: 5142248 (5143590)
W0601 11:51:33.713162       1 reflector.go:289] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:417: watch of *v1.ConfigMap ended with: too old resource version: 5143997 (5145826)
```

**Anything else we need to know?**:

**Comment by montenegrodr** [comment]
Hold on, I guess the problem isn't prometheus operator, but the service itself. I'll close this issue in a minute if that is true.

**Comment by montenegrodr** [comment]
Closing the issue since it was unrelated to prometheus operator. 

For posterity:

I found a bit hard to troubleshoot. I figure out what happened by visiting `http://<prometheus>:9090/targets`. Saw there the error, which was:

```
Get http://10.20.0.46:9292/metrics: dial tcp 10.20.0.46:9292: connect: connection refused kubernetes
```

So the service was unreachable.. I took from granted it was working since `kubectl port-forward` worked. But the address was not actually bound to the private ip. I just needed to fix it in the container.

Thanks anyway.
